File: algorithms/sorting/odd-even_sort/odd-even_sort.php
<?php

function oddEvenSort($l)
{
    $lc = array_replace([], $l);
    $s = false;
    while (!$s) {
        $s = true;
        for ($i = 1; $i < count($lc) - 2; ++$i)
            if ($lc[$i] > $lc[$i + 1]) {
                $tmp = $lc[$i];
                $lc[$i] = $lc[$i + 1];
                $lc[$i + 1] = $tmp;
                $s = false;
            }
        for ($i = 0; $i < count($lc) - 1; ++$i)
            if ($lc[$i] > $lc[$i + 1]) {
                $tmp = $lc[$i];
                $lc[$i] = $lc[$i + 1];
                $lc[$i + 1] = $tmp;
                $s = false;
            }
    }

    return $lc;
}

$list = [9, 8, 7, 6, 5, 4, 3, 2, 1, 0];

echo "Input:\n";
print_r($list);

echo "\n";

echo "Output:\n";
print_r(oddEvenSort($list));

File: algorithms/sorting/odd-even_sort/odd-even_sort.c
#include <stdbool.h>
#include <stdio.h>

void printAray(int arr[], int n) {
  int x = 0;
  while (x < n) {
    printf("%i,", arr[x]);
    ++x;
  }
  printf("\n");
}

void handleSwap(int arr[], int i, bool *isSorted) {
  if (arr[i] > arr[i + 1]) {
    int tmp = arr[i];
    arr[i] = arr[i + 1];
    arr[i + 1] = tmp;
    *isSorted = false;
  }
}

void oddEvenSort(int arr[], int n) {
  bool isSorted = false;

  while (!isSorted) {
    isSorted = true;

    for (int i = 1; i < n - 1; i = i + 2)
      handleSwap(arr, i, &isSorted);
    for (int i = 0; i < n - 1; i = i + 2)
      handleSwap(arr, i, &isSorted);
  }
}

int main() {
  int arr[] = {9, 8, 7, 6, 5, 4, 3, 2, 1, 0};
  int n = sizeof(arr) / sizeof(arr[0]);
  printf("Input\n");
  printAray(arr, n);
  oddEvenSort(arr, n);
  printf("Output\n");
  printAray(arr, n);
  return 0;
}
